argoverse/utils/make_att_files.py

```diff
@@ -0,0 +1,365 @@
+                    data_amodel[key]["track_label_frames"] = dict_track_labels[key]
+
+            argoverse_loader = ArgoverseTrackingLoader(os.path.join(root_dir, name_folder))
+            data_log = argoverse_loader.get(id_log)
+            list_lidar_timestamp = data_log.lidar_timestamp_list
+
+            dict_tracks: Dict[str, Any] = {}
+            for id_track in data_amodel.keys():
+
+                data = data_amodel[id_track]
+                if data["label_class"] not in list_name_class:
+                    continue
+
+                data_per_frame = data["track_label_frames"]
+
+                dict_per_track: Dict[str, Any] = {}
+                dict_tracks[id_track] = dict_per_track
+                dict_tracks[id_track]["ind_lidar_min"] = -1
+                dict_tracks[id_track]["ind_lidar_max"] = -1
+                length_log = len(list_lidar_timestamp)
+                dict_tracks[id_track]["list_city_se3"] = [None] * length_log
+                dict_tracks[id_track]["list_bbox"] = [None] * length_log
+                count_track += 1
+
+                dict_tracks[id_track]["list_center"] = np.full([length_log, 3], np.nan)
+                dict_tracks[id_track]["list_center_w"] = np.full([length_log, 3], np.nan)
+                dict_tracks[id_track]["list_dist"] = np.full([length_log], np.nan)
+                dict_tracks[id_track]["exists"] = np.full([length_log], False)
+
+                for box in data_per_frame:
+
+                    if box["timestamp"] in list_lidar_timestamp:
+                        ind_lidar = list_lidar_timestamp.index(box["timestamp"])
+                    else:
+                        continue
+
+                    if dict_tracks[id_track]["ind_lidar_min"] == -1:
+                        dict_tracks[id_track]["ind_lidar_min"] = ind_lidar
+
+                    if (
+                        dict_tracks[id_track]["ind_lidar_max"] == -1
+                        or ind_lidar > dict_tracks[id_track]["ind_lidar_max"]
+                    ):
+                        dict_tracks[id_track]["ind_lidar_max"] = ind_lidar
+
+                    center = np.array([box["center"]["x"], box["center"]["y"], box["center"]["z"]])
+                    egovehicle_to_city_se3 = argoverse_loader.get_pose(ind_lidar, id_log)
+                    if egovehicle_to_city_se3 is None:
+                        print("Pose not found!")
+                        continue
+                    center_w = egovehicle_to_city_se3.transform_point_cloud(center[np.newaxis, :])[0]
+
+                    dict_tracks[id_track]["list_center"][ind_lidar] = center
+                    dict_tracks[id_track]["list_center_w"][ind_lidar] = center_w
+                    dict_tracks[id_track]["list_dist"][ind_lidar] = np.linalg.norm(center[0:2])
+                    dict_tracks[id_track]["exists"][ind_lidar] = True
+                    dict_tracks[id_track]["list_city_se3"][ind_lidar] = egovehicle_to_city_se3
```

Comment:
I usually used the notation `city_SE3_egovehicle` instead of `egovehicle_to_city_se3` just so that the semantics are
`p_city = city_SE3_egovehicle * p_egovehicle`. You are expressing the transformation from the egovehicle frame, to the city, right? I think something similar is often [done in GTSAM too](https://gtsam.org/gtsam.org/2020/06/28/gtsam-conventions.html)